### PR TITLE
feat: add YAML and JSON support for bundles

### PR DESCRIPTION
**Pull request by b4nst**
This little addition add YAML support for bundle file. It can be useful for multiple purposes, like using `sops` encrypted files until we address #74.

I'm also referencing cue-lang/cue/discussions/2452 in case a better way of including Orphan files exists.

### CODE REVIEW

**Review by stefanprodan — COMMENTED** [review]
Hey @b4nst we have similar feature for [values](https://github.com/stefanprodan/timoni/blob/main/cmd/timoni/build.go#L205), where we allow both YAML and JSON. I think we need to support JSON as well for consistency.

Also a test with a bundle made out of 3 files CUE, YAML and JSON would be good.

**Comment by b4nst** [comment]
Hey @stefanprodan, just added JSON support and a test case with the combo of all file types.

**Comment by stefanprodan** [comment]
Thanks @b4nst can you please rebase with main, solve conflicts and force push to unblock CI.

**Comment by b4nst** [comment]
Of course! Sorry I went a bit fast on the push 😅

**Comment by b4nst** [comment]
Ok rebased, seems there's an issue with the Copy and YAML files during workspace creation introduced with the injection system (break the content of the file). I'll give it a look

**Comment by stefanprodan** [comment]
@b4nst I suggest we put in the workspace not the jsons and yamls but their cue value.

**Comment by b4nst** [comment]
Yup, that's what I started (moving the extract part into the workspace function). Incoming stoday

File: internal/engine/bundle_builder.go
Comment (stefanprodan) [inline internal/engine/bundle_builder.go:82]: This approach has many issues, please let's look at the file extension in a switch statement instead of hiding errors. If a cue, json or yaml is invalid, we should tell users and exit, instead of silently ignoring errors which would result in applying something on a cluster that was never intended.

Comment (b4nst) [reply internal/engine/bundle_builder.go:82]: Okay with me, but it means we don't support file with incorrect extension (which is probably a fair assumption).

Comment (stefanprodan) [reply internal/engine/bundle_builder.go:82]: Yes we don't support incorrect extension, the switch should match:  https://github.com/stefanprodan/timoni/blob/main/cmd/timoni/build.go#L205

Comment (b4nst) [reply internal/engine/bundle_builder.go:82]: Perfect, going to grab some food and pushing you a fix after

Comment (stefanprodan) [reply internal/engine/bundle_builder.go:82]: Thanks

Comment (b4nst) [reply internal/engine/bundle_builder.go]: Okay did a switch around extensions. Also refactored the injector so it harmonizes the pattern and avoid double parsing / writing of files. I also dropped the copy dependency in the meantime.

Comment (stefanprodan) [reply internal/engine/bundle_builder.go]: Looks great, thanks a lot @b4nst.

I was wandering if we can create the workspace in-memory instead of on disk in tmp. I haven't figure out how to pass bytes to cue build, but it must be a way.

File: cmd/timoni/bundle_build_test.go
Comment (hiddeco) [inline cmd/timoni/bundle_build_test.go:80]: `ioutil` is deprecated since Go 1.17, this should be `os.CreateTemp` from top of head.

Comment (stefanprodan) [reply cmd/timoni/bundle_build_test.go:80]: Good catch, should be like we do it elsewhere:

```
	bundlePath := filepath.Join(t.TempDir(), "bundle.cue")
	err = os.WriteFile(bundlePath, []byte(bundleData), 0644)
	g.Expect(err).ToNot(HaveOccurred())
```

`t.TempDir()` does a cleanup so we don't leave stuff in tmp.

Comment (b4nst) [reply cmd/timoni/bundle_build_test.go:80]: Oh nice catch! Will address it, thanks

File: internal/engine/bundle_builder.go
Comment (stefanprodan) [reply internal/engine/bundle_builder.go]: New PR for sure, after we get this in shape and merged. Thanks for looking into it.

File: cmd/timoni/bundle_build_test.go
Comment (b4nst) [reply cmd/timoni/bundle_build_test.go]: Done

File: internal/engine/bundle_builder.go
Comment (b4nst) [reply internal/engine/bundle_builder.go]: We might want to revive https://github.com/cue-lang/cue/issues/607 for a proper in-memory support

File: cmd/timoni/bundle_build_test.go
Comment (hiddeco) [inline cmd/timoni/bundle_build_test.go:81]: Every time `t.TempDir()` is called, a new temporary directory is created from what I recall. I would therefore call it once and then re-use it.

Comment (hiddeco) [reply cmd/timoni/bundle_build_test.go:81]: > Each subsequent call to t.TempDir returns a unique director
> &mdash; https://pkg.go.dev/testing#T.TempDir

Comment (b4nst) [reply cmd/timoni/bundle_build_test.go:81]: Hum indeed. I don't think it will drastically improve perf but that would be cleaner indeed. Wilco

**Review by hiddeco — APPROVED** [review]
This looks good to me :100:

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @b4nst 🥇

**Pull request merged**
